graviti/manager/tag.py

```diff
@@ -51,34 +64,55 @@ def create(self, name: str, revision: Optional[str] = None) -> TensorbayTag:
                 the branch name, or the tag name.
                 If the revision is not given, create the tag for the current commit.
 
-        Return:
-            The :class:`.TensorbayTag` instance with the given name.
+        Returns:
+            The :class:`.Tag` instance with the given name.
 
         """
+        if not revision:
+            revision = self._commit_id
 
-    def get(self, name: str) -> TensorbayTag:
+        response = create_tag(**self._access_info, name=name, revision=revision)
+        return Tag.from_pyobj(response)
+
+    def get(self, name: str) -> Tag:
         """Get the certain tag with the given name.
 
         Arguments:
             name: The required tag name.
 
-        Return:
-            The :class:`.TensorbayTag` instance with the given name.
+        Raises:
+            TypeError: When the given tag name is illegal.
+
+        Returns:
+            The :class:`.Tag` instance with the given name.
 
         """
+        if not name:
+            raise TypeError("The given tag name is illegal")
+
+        response = get_tag(**self._access_info, tag=name)
+        return Tag.from_pyobj(response)
 
-    def list(self) -> PagingList[TensorbayTag]:
+    def list(self) -> PagingList[Tag]:
         """List the information of tags.
 
-        Return:
-            The PagingList of :class:`tags<.TensorbayTag>` instances.
+        Returns:
+            The PagingList of :class:`tags<.Tag>` instances.
 
         """
+        return PagingList(self._generate, 128)
 
     def delete(self, name: str) -> None:
         """Delete a tag.
 
         Arguments:
             name: The tag name to be deleted for the specific commit.
 
+        Raises:
+            TypeError: When the given tag name is illegal.
+
         """
+        if not name:
```

Comment:
why need to check name here?
can OpenAPI check this for us?

Reply:
because here the name is the part of url. If the name = "", the url will be change, then backend will raise ResourceNotExist error about the url instead of the ResourceNotExist about tag

Reply:
OK,
1. Should we move the check to `openapi` level?
2. Should we also check the name is `str` or not?

Reply:
1. yes
2. No, in the openapi, we use f string. Here we remind the user of the variable type, and even if the user set name = 1. The openapi will not raise error. I think this way is ok.